storage/provider/openstack.go

```diff
@@ -0,0 +1,328 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package provider
+
+import (
+	"math"
+	"net/url"
+
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/storage"
+
+	"github.com/juju/errors"
+	"github.com/juju/names"
+
+	"gopkg.in/goose.v1/cinder"
+	"gopkg.in/goose.v1/client"
+	"gopkg.in/goose.v1/identity"
+	"gopkg.in/goose.v1/nova"
+)
+
+const (
+	OpenstackProviderType = storage.ProviderType("openstack")
```

Comment:
This should be named CinderProviderType
Just as for AWS, we have a EBS_Provider type. 
The storage provider type is named after the type of volume it manages, not the cloud on which it provisions the storage.